meta-xt-driver-domain-virtio/recipes-extended/xen/xen-tools_git.bbappend

```diff
@@ -0,0 +1,16 @@
+FILESEXTRAPATHS:prepend := "${THISDIR}/files:"
+
+FILES:${PN}-devd += " \
+${sysconfdir}/systemd/system/xendriverdomain.service.d/weston-env.conf \
+"
+
+do_install:append() {
+    install -d ${D}${sysconfdir}/systemd/system/xendriverdomain.service.d
+    install -m 0644 ${WORKDIR}/weston-env.conf ${D}${sysconfdir}/systemd/system/xendriverdomain.service.d
+}
+
+python () {
+    src_uri_append = "file://weston-env.conf"
+    if src_uri_append not in d.getVar('SRC_URI'):
```

Comment:
Hi @lorc, in the driver domain we have the following order of the layers application:

```
  /mnt/build/prod-devel-rcar/dev/vladyslav-goncharuk/virtio-integration/yocto/meta-xt-common/meta-xt-driver-domain-virtio \
...
  /mnt/build/prod-devel-rcar/dev/vladyslav-goncharuk/virtio-integration/yocto/meta-xt-prod-devel-rcar/meta-xt-domx-gen3 
``` 

meta-xt-prod-devel-rcar/meta-xt-domx-gen3 layer has the xen-source.inc file, which unconditionally overrides the SRC_URI - https://github.com/xen-troops/meta-xt-prod-devel-rcar/blob/4c277f4f00095bb6875a1d82b51c63ef74704bbd/meta-xt-domx-gen3/recipes-extended/xen/xen-source.inc#L3

And that is done for a valid reason. All files from the original recipe SHOULD be dropped.

But we still need to inject weston-env.conf in case of virtio build from the meta-xt-common/meta-xt-driver-domain-virtio layer in case of virtio build. And that should be done unconditionally from what is done in more specific layers. Moving this part to meta-xt-prod-devel-rcar is wrong from my point of view, as it is a common setting for all virtio builds.

So this hack fights with a contradiction.

Reply:
Well, problem is that you are introducing some knowledge about other layers. As meta-xt-common is the base layer, it should know nothing about subsequent platform and product layers. And of course it should not employ some quirks to workaround what other layers do.

Reply:
Hi @lorc, done.